### PR TITLE
Release 4.3.0

### PR DESCRIPTION
**Pull request by SylvainCorlay**
I am also 
- adding myself as a recipe maintainer.
- using a sha256 hash instead of md5.
- requiring six as a dependency
- adding enum34 as a dependency in the case of python 2.x and 3.3
- re-rendering the feestock.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,10 +32,11 @@ test:
...
   license: BSD 3-clause
-  summary: configuration system for Python applications
+  summary: Configuration system for Python applications
 
 extra:
   recipe-maintainers:
     - jakirkham
     - pelson
     - minrk
+    - scorlay
```
Comment (jakirkham) [inline recipe/meta.yaml]: SylvainCorlay

Comment (SylvainCorlay) [inline recipe/meta.yaml]: good catch :+1:

**Comment by SylvainCorlay** [comment]
too bad we cannot cancel appveyor build that are still in the queue.

**Comment by ocefpaf** [comment]
> too bad we cannot cancel appveyor build that are still in the queue.

I cancelled those for you but thanks to [these lines](https://github.com/conda-forge/traitlets-feedstock/blob/master/appveyor.yml#L52-L55) a new commit forces the old `AppVeyor` build to exit early and reduce the amount of time we have to wait.

**Comment by SylvainCorlay** [comment]
Ready to go!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,20 +8,21 @@ package:
 source:
   fn: traitlets-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/t/traitlets/traitlets-{{ version }}.tar.gz
-  md5: d0725f9818bd9e112a6531da26d433b8
+  sha256: {{ sha256 }}
 
 build:
   number: 0
-  script: pip install --no-deps .
+  script: python setup.py install
```
Comment (minrk) [inline recipe/meta.yaml]: This shouldn't have changed

Comment (ocefpaf) [inline recipe/meta.yaml]: @SylvainCorlay note that we want to go in the `pip install --no-deps .` direction everywhere. We are just waiting for some fixes in `conda-build` to do so. There are many advantages and, IMO as a packager, the best one is to get the `entry_points` for free.

Comment (SylvainCorlay) [inline recipe/meta.yaml]: I understand but `pip` is not found at the moment.

Comment (ocefpaf) [inline recipe/meta.yaml]: That another issue. While PyPA is moving towards making `pip` and `setuptool` one with the basic python distribution conda-forge went in the opposite direction by removing them.

So you need to add `pip` as a build dependency.

Comment (minrk) [inline recipe/meta.yaml]: @ocefpaf is there a link to the discussion of removing pip from Python in conda-forge? I see that `conda install python` still gets pip locally when everything's coming from conda-forge, so it seems weird that the dependencies in the recipe resolve differently than they would for a normal conda env.

anaconda-installed Python lacked pip in the beginning, which caused endless headaches and user problems, which eventually got fixed when `conda install python` included pip as users expect. I would like to make sure that conda-forge doesn't make the same mistake again.

**Comment by SylvainCorlay** [comment]
Finally!

**Comment by minrk** [comment]
Thanks, @SylvainCorlay!

File: recipe/meta.yaml (hunk 2 above)
Comment (ocefpaf) [inline recipe/meta.yaml]: >  is there a link to the discussion of removing pip from Python in conda-forge?

That is scattered in many places but maybe @jjhelmus, who is our python-feedstock hero, can answer that.

> anaconda-installed Python lacked pip in the beginning,

> I would like to make sure that conda-forge doesn't make the same mistake again.

:+1: to that. I am trying to be vocal on that but I am only one. Maybe we could write an enhancement proposal to explain the prons and cons and try to  move towards the easier `pip install --no-deps .`

We have a meeting today and I will bring this issue.

**Comment by jakirkham** [comment]
This has come up before and been answered elsewhere. In order to package `pip` and `setuptools` (possibly `wheel` IIRC), one cannot have `pip` and `setuptools` in the environment. As the flag to do this is global, this affects everything. We also wanted to provide people that optionally installed with ``setuptools`` the ability to use ``distutils`` if they wanted. Should add this is also the same setup used by Continnum when building. So there is some value in following their lead.

Switching this back globally would be a big problem if we couldn't disable it locally for those recipes needing too. It looks like `conda-build` will get the option to disable this locally. So we can revisit when that is in a release we are using.

**Comment by minrk** [comment]
Thanks, @jakirkham! As long as `conda install python` always gets pip, I don't think there's a big issue.

**Comment by jakirkham** [comment]
Right, so this setting only affects our build infrastructure. It has no effect on anyone's own infrastructure. Pulling in `pip` and `setuptools` will always be the default.

**Comment by jakirkham** [comment]
Also some xref's

* `pip`was added in PR ( https://github.com/conda-forge/staged-recipes/pull/864 ).
* `setuptools`was added in PR ( https://github.com/conda-forge/staged-recipes/pull/749 ).
* Was wrong about `wheel`, it uses `setuptools` for the build.
* Finally PR ( https://github.com/conda/conda-build/pull/1311 ) introduces the setting to optionally disable `pip` and `setuptools` during a build.

**Comment by SylvainCorlay** [comment]
btw I don't think I was invited to the repo as a consequence of the recipe maintainer list being changed.

**Comment by jakirkham** [comment]
When you add yourself to an existing recipe, it takes roughly 24-hrs to go through. The reason being there is a maintenance job that scans every feedstock in conda-forge and makes a global update of this stuff. If you don't see it by this time tomorrow, definitely let us know.

**Comment by jjhelmus** [comment]
Should add a link to the conda's [`add_pip_as_python_dependency` configuration option](http://conda.pydata.org/docs/config.html#add-pip-as-python-dependency-add-pip-as-python-depedency) which controls if conda installs `python+pip+setuptools` or just `python` when python is an install target or requirement.

**Comment by jakirkham** [comment]
It looks like you got added, @SylvainCorlay. So I think we are in good shape.

**Comment by SylvainCorlay** [comment]
Thank you I did not know it was a cron job hence my question.

**Pull request merged**